src/bp-core/bp-core-functions.php

```diff
@@ -4937,7 +4931,7 @@ function( $file ) {
 		$latest_deprecated_functions_versions = array_splice( $latest_deprecated_functions_versions, -2 );
 	}
 
-	$index_initial_version = array_search( $first_major, $latest_deprecated_functions_versions, true );
+	$index_initial_version = array_search( $initial_version, $latest_deprecated_functions_versions, true );
```

Comment:
And,, yet, another **`$initial_version`** is a float value and **`$latest_deprecated_functions_versions`** is an array of strings (looks like it would be a single element with a value of "12.0", if I'm not mistaken).

Reply:
$latest_deprecated_functions_versions_float = array_map('floatval', $latest_deprecated_functions_versions);
$index_initial_version = array_search($initial_version, $latest_deprecated_functions_versions_float, true);

Instead of making these point-of-use changes, we can also initially check and make the $deprecated_functions_versions array float.

Reply:
True about apples to apples comparisons, at some point the types should be made the same.

Reply:
It should be ok now the list of versions is an array of float values.